packages/espressocash_app/lib/features/incoming_single_link_payments/src/widgets/link_listener.dart

```diff
@@ -35,10 +34,11 @@ class _ISLPListenerState extends State<ISLPListener> {
   }
 
   Future<void> _processISLP(SingleKeyPaymentData data) async {
-    final id = await context.createISLP(data);
+    // Added this delay to avoid routing error
+    await Future<void>.delayed(const Duration(milliseconds: 100));
```

Comment:
This looks like a hack. Why are these delays needed?

Reply:
Removed the delays. It was related to routing. Moved initialization of link listeners after router is created